File: backend/GridEventManagement/Models/TodoItem.cs
using GridEventManagement.Web.Enums;

namespace GridEventManagement.Web.Models;

public class TodoItem
{
    public int Id { get; set; }
    public int UserId { get; set; }
    public TodoType Type { get; set; }
    public int RelatedId { get; set; }
    public string Title { get; set; } = string.Empty;
    public DateTime? DueDate { get; set; }
    public bool IsCompleted { get; set; }
    public DateTime? CompletedAt { get; set; }

    public User User { get; set; } = null!;
}

File: backend/src/main/java/com/pm/workstation/entity/TodoItem.java
package com.pm.workstation.entity;

import com.pm.workstation.enums.TodoStatus;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "todo_item")
public class TodoItem {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private Long requirementId;

    @Column(nullable = false)
    private Long userId;

    @Column(nullable = false, length = 256)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private TodoStatus status;

    @Column
    private LocalDate dueDate;

    @Column
    private Long reminderRuleId;

    @Column(nullable = false)
    private Boolean reminded;

    @Column
    private LocalDateTime completedAt;

    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @Column(nullable = false)
    private LocalDateTime updatedAt;
}
